Remove CSSKeyframesRule.insertRule()

karlcow opened the pull request:
#### c26a65e62e3999f70665e1aaa493acc9938891ae
<pre>
Remove CSSKeyframesRule.insertRule()
<a href="https://bugs.webkit.org/show_bug.cgi?id=247364">https://bugs.webkit.org/show_bug.cgi?id=247364</a>
rdar://102141002

Reviewed by Antoine Quint.

Removes insertRule. It was removed from Gecko, 10 years ago.
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=841896">https://bugzilla.mozilla.org/show_bug.cgi?id=841896</a>
It was first deprecated in Blink and removed from Blink 7 years ago.
<a href="https://groups.google.com/a/chromium.org/g/blink-dev/c/wW7w8Hro78s/m/JDULF2oM9iIJ">https://groups.google.com/a/chromium.org/g/blink-dev/c/wW7w8Hro78s/m/JDULF2oM9iIJ</a>
GitHub Search doesn&apos;t return meaningful instances of insertRule.

* LayoutTests/animations/change-keyframes-expected.txt:
* LayoutTests/animations/change-keyframes.html:
* LayoutTests/animations/CSSKeyframesRule-parameters-expected.txt:
* LayoutTests/animations/CSSKeyframesRule-parameters.html:
* LayoutTests/http/tests/css/resources/shared-stylesheet-mutation.js:
(executeTests):
* LayoutTests/http/tests/css/shared-stylesheet-mutation-preconstruct-expected.txt:
* Source/WebCore/css/CSSKeyframesRule.cpp:
(WebCore::CSSKeyframesRule::insertRule): Deleted.
* Source/WebCore/css/CSSKeyframesRule.h:
* Source/WebCore/css/CSSKeyframesRule.idl:

Canonical link: <a href="https://commits.webkit.org/258361@main">https://commits.webkit.org/258361@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bdd3e8a66a9f0ace2747ecf5a222ccfbb56df1f1

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/101643 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/10794 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/34710 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/110976 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/171179 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/105622 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/11750 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/1704 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/94046 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/108738 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/107423 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/8966 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/92221 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/36295 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/90859 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/23642 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/78520 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/4392 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/25139 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/4465 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/1592 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/10553 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/44627 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/6221 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/3023 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

graouts reviewed (approved):
I don’t understand the changes made to LayoutTests/http/tests/css/resources/shared-stylesheet-mutation.js but the source changes look good.

karlcow committed:
Remove CSSKeyframesRule.insertRule()
https://bugs.webkit.org/show_bug.cgi?id=247364
rdar://102141002

Reviewed by Antoine Quint.

Removes insertRule. It was removed from Gecko, 10 years ago.
https://bugzilla.mozilla.org/show_bug.cgi?id=841896
It was first deprecated in Blink and removed from Blink 7 years ago.
https://groups.google.com/a/chromium.org/g/blink-dev/c/wW7w8Hro78s/m/JDULF2oM9iIJ
GitHub Search doesn't return meaningful instances of insertRule.

* LayoutTests/animations/change-keyframes-expected.txt:
* LayoutTests/animations/change-keyframes.html:
* LayoutTests/animations/CSSKeyframesRule-parameters-expected.txt:
* LayoutTests/animations/CSSKeyframesRule-parameters.html:
* LayoutTests/http/tests/css/resources/shared-stylesheet-mutation.js:
(executeTests):
* LayoutTests/http/tests/css/shared-stylesheet-mutation-preconstruct-expected.txt:
* Source/WebCore/css/CSSKeyframesRule.cpp:
(WebCore::CSSKeyframesRule::insertRule): Deleted.
* Source/WebCore/css/CSSKeyframesRule.h:
* Source/WebCore/css/CSSKeyframesRule.idl:

Canonical link: https://commits.webkit.org/258361@main
